### ISSUE TITLE
A small patch to incorporate from the Fixpack

### ISSUE THREAD

**Issue by agb1** [RoxanneSHS/ChloeEET#1]
In the setup-tp2, replace the four instances of:

  WRITE_BYTE    0x246 ~%YRKENSAI%~
  WRITE_BYTE    0x247 0x40

with:

  WRITE_SHORT   0x246 0x4000 + ~%YRKENSAI%~

This was done for TobEx compatibility but the same is appropriate for EE, I think.

**Comment by RoxanneSHS** [comment]
WRITE_SHORT 0x244 0
WRITE_SHORT   0x246 0x4000 + ~%CVMystra%~
WRITE_BYTE  0x247 0x40

This is what I use for Sandrah since the suggested one never worked (this issue has been around for some time and it has been advocated by some and I never saw it functioning.) 
The old code still seems to work well. Did you actually test it or is it just found by cleaning up all the stuff you do?
Not sure if a change here is really an improvement.

**Comment by agb1** [comment]
This is an old patch; I haven't tested it.  But if I'm interpreting it correctly, the problem it solves would arise if Chloe was installed late in the install order, after many other kits had been added.  To be exact, the former code would fail if the value of YRKENSAI exceeds 255 (if some 200ish kits were installed before Chloe, since the original kits already take up some slots), since 255 is the maximum that can be written as a single byte.  The replacement code adds the position of the new kit to 0x4000 and writes the result as a single value, so it can accept values of YRKENSAI up to 0xFFFF - 0x4000 = 49,151 kits.

**Comment by agb1** [comment]
Just as a point of reference, my current EET install has 86 entries in kitlist.2da, and it's not particularly heavy on kit mods.  BWS installs Chloe early, so she's not at risk in BWS installs; it's really for manual installs and general "correctness."  Since Sandrah is installed late, though, you should make sure to use the WRITE_SHORT method for her kit.

**Comment by agb1** [comment]
Also, to be clear, the problem solved by this code is that the cre file would be given a different kit number than intended.  Instead of pointing to YRKENSAI, the cre would point to some early kit due to wraparound.

**Comment by RoxanneSHS** [comment]
I know all of this was a BGT and ToBex issue.
I may be wrong, but I had the impression that this is no longer a problem with EE v2.3.

Anyway, I plan a re-install later this day and I may include this for testing (before I comit anything to github.)

**Comment by agb1** [comment]
I will look into this.  If the proposed code has adverse effects then many other mods that use this pattern will need to be patched for EE.

**Comment by agb1** [comment]
The modified code seems to work in-game on BG2EE:  https://s16.postimg.org/hhzm7zjyt/Baldr000.jpg

Have not yet tried it on EET.

**Comment by agb1** [comment]
Just to check:  you kept the WRITE_SHORT 0x244 0, right?

So:

WRITE_SHORT 0x244 0
WRITE_SHORT 0x246 0x4000 + %YRKENSAI%

**Comment by RoxanneSHS** [comment]
WRITE_SHORT 0x244 0
WRITE_SHORT   0x246 0x4000 + ~%CVMystra%~
This is what I used for Sandrah (not comparable however, since the kit is only affecting her second class, i.e. she is a Fighter/Priestess of Mystra

and I used similar for Chloe
I have the display you attached when I use *old* code and I have *Fighter* with new code. Problem may well be my method of testing, maybe?
I am however convinced that your finding is reliable and thus will update Chloe with it.
